.github/workflows/semantic-release.yml

```diff
@@ -4,35 +4,63 @@ on:
   push:
     # These are the branches that we expect to publish from
     # semantic-release also has a regex that validates we're running on a release branch
-    branches:    
+    branches:
       - master
-      - '[0-9]+.x'
-      - '[0-9]+.x.x'
-      - '[0-9]+.[0-9]+.x'
+      - "[0-9]+.x"
+      - "[0-9]+.x.x"
+      - "[0-9]+.[0-9]+.x"
 
 jobs:
   release:
     runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.versionNumber.outputs.version }}
     steps:
-    - uses: actions/checkout@v2
-      with:
-        fetch-depth: 0
-        persist-credentials: false
-    - uses: actions/setup-node@v1
-      with:
-        node-version: '>=14.16.0 14'
-    - run: npm ci
-    - name: Import GPG key
-      uses: crazy-max/ghaction-import-gpg@v3.0.1
-      with:
-        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
-        passphrase: ${{ secrets.GPG_PASSPHRASE }}
-        git-user-signingkey: true
-        git-commit-gpgsign: true
-    - run: npx semantic-release
-      env:
-        # A personal access token is required to publish from protected branches
-        GITHUB_TOKEN: ${{ secrets.SEMANTIC_RELEASE_TOKEN }}
-        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
-        GIT_COMMITTER_NAME: Sage Carbon
-        GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMITTER_EMAIL }} 
\ No newline at end of file
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ">=14.16.0 14"
+      - run: npm ci
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v3.0.1
+        with:
+          gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git-user-signingkey: true
+          git-commit-gpgsign: true
+      - run: npx semantic-release
+        env:
+          # A personal access token is required to publish from protected branches
+          GITHUB_TOKEN: ${{ secrets.SEMANTIC_RELEASE_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          GIT_COMMITTER_NAME: Sage Carbon
+          GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMITTER_EMAIL }}
+      - id: versionNumber
+        run: echo "::set-output name=version::$SEMVER_VERSION"
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: release
+    if: ${{ needs.release.outputs.version != null }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ">=14.16.0 14"
+      - run: npm ci
+      - name: Generate metadata file
+        run: npm run generate-metadata
```

Comment:
Please can we verify that if the npm registry is down that this exits with an error code that would cancel the build.